tests/functional/test_upload.py

```diff
@@ -275,6 +275,31 @@ def test_upload_for_fileobj(self):
         self.assert_expected_client_calls_were_correct()
         self.assert_upload_part_bodies_were_correct()
 
+    def test_limits_in_memory_chunks_for_fileobj(self):
+        # Limit the maximum in memory chunks to one but make number of
+        # threads more than one. This means that the upload will have to
+        # happen sequentially despite having many threads available because
+        # data is sequentially partitioned into chunks in memory and since
+        # there can only every be one in memory chunk, each upload part will
+        # have to happen one at a time.
```

Comment:
Let me phrase it another way.  Where in this test method is that expectation called out?  That "if there is a discrepancy then we'll fail in the stubber."  Is that implicit in the sequencing of calls to `self.add_*` below?

If so, then I think that's pretty reasonable.